sync-core/src/main/java/com/cloudant/sync/datastore/encryption/EncryptedAttachmentInputStream.java

```diff
@@ -0,0 +1,259 @@
+     * @throws IOException
+     */
+    public int read() throws IOException {
+        return cipherInputStream.read();
+    }
+
+    @Override
+    /**
+     * <p>Reads up to b.length bytes of data from this input stream into an array of bytes.</p>
+     *
+     * <p>This method proxies to the CipherInputStream.</p>
+     *
+     * @param b the buffer into which the data is read
+     *
+     * @return the total number of bytes read into the buffer, or -1 is there is no more data
+     *          because the end of the stream has been reached
+     * @throws IOException
+     */
+    public int read(byte[] b)
+            throws IOException {
+        return cipherInputStream.read(b);
+    }
+
+    @Override
+    /**
+     * <p>Reads up to len bytes of data from this input stream into an array of bytes. This
+     * method blocks until some input is available. If the first argument is null, up to len
+     * bytes are read and discarded.</p>
+     *
+     * @param b the buffer into which the data is read.
+     * @param off the start offset in the destination array buf
+     * @param len the maximum number of bytes read.
+     * @return the total number of bytes read into the buffer, or -1 if there is no more data
+     *          because the end of the stream has been reached.
+     * @throws IOException
+     */
+    public int read(byte[] b,
+                    int off,
+                    int len)
+            throws IOException {
+        return cipherInputStream.read(b, off, len);
+    }
+
+    @Override
+    /**
+     * <p>Skips n bytes of input from the bytes that can be read from this input stream
+     * without blocking.</p>
+     *
+     * <p>Fewer bytes than requested might be skipped. The actual number of bytes skipped is
+     * equal to n or the result of a call to available, whichever is smaller. If n is less
+     * than zero, no bytes are skipped.</p>
+     *
+     * <p>The actual number of bytes skipped is returned.</p>
+     *
+     * @param n the number of bytes to be skipped.
+     * @return the actual number of bytes skipped.
+     * @throws IOException
+     */
+    public long skip(long n)
+            throws IOException {
+        return cipherInputStream.skip(n);
+    }
+
+    @Override
+    /**
+     * <p>Returns the number of bytes that can be read from this input stream without
+     * blocking. The available method of InputStream returns 0. This method should be
+     * overridden by subclasses.</p>
+     *
+     * @return the number of bytes that can be read from this input stream without blocking.
+     * @throws IOException
+     */
+    public int available()
+            throws IOException {
+        return cipherInputStream.available();
+    }
+
+    @Override
+    /**
+     * <p>Closes this input stream and releases any system resources associated with the stream.</p>
+     *
+     * <p>The EncryptedAttachmentInputStream method of CipherInputStream calls the close method
```

Comment:
I find this comment pretty confusing. Wouldn't it be easier just to say this method calls close on its proxied CipherInputStream?